milli/src/vector/parsed_vectors.rs

```diff
@@ -61,7 +71,13 @@ impl ParsedVectorsDiff {
                 return Err(error);
             }
         }
-        .flatten();
+        .flatten().map_or(BTreeMap::default(), |del| del.into_iter().map(|(name, vec)| (name, Some(vec))).collect());
+        for embedding_config in embedders_configs {
+            if embedding_config.user_defined.contains(docid) {
+                old.entry(embedding_config.name.to_string()).or_insert(None);
```

Comment:
It is not clear to me what it means to have None here.

- When does it happen?
- what should calling code do when receiving None as value for an embedder?

I'm guessing it is when we had vectors for the embedder but can't get them because no db.

Can we indicate this somewhere as a comment?